horace_core/utilities/split_vector.m

```diff
@@ -0,0 +1,100 @@
+function [chunks, idxs] = split_vector(vector, max_chunk_sum)
+%SPLIT_VECTOR Split the given vector into a set of sub-vectors such that the
+% sum of each sub-vector has a maximum of max_chunk_sum, or the sub-vector has length
+% 1.
+%
+% If a value in `vector` is greater than max_chunk_sum, then that value will comprise
+% its own sub-vector.
+%
+% Input:
+% ------
+% vector       A vector of non-negative values.
+% max_chunk_sum      A positive value specifying the maximum sum for each sub-vector.
+%
+% Output:
+% -------
+% chunks       Cell array of vectors. Concatenation of these vectors will be
```

Comment:
it's not an array of vectors, it's a cell array of arrays; if you're using vector as a 'type' here that makes the argument name even more confusing

Reply:
Again, given you can do `validateattributes(x, {'numeric'}, {'vector'}`, I think this is well defined. Perhaps label that it's a numeric vector, though.

Reply:
A "vector" in Matlab is a [1, n] or [n, 1] array. So it's not a type (or a reserved name), it's a commonly used name for a special case of `array`. See the Matlab function `isvector`.